<!-- -*- mode: XML -*- -->
<launch>
  <arg name="model" default="$(find akrobat)/urdf/hexapod.urdf" />
  <arg name="gui" default="true" />
  <arg name="dynamixel_namespace" value="dynamixel_controller" />

  <param name="robot_description" textfile="$(arg model)" />
  <param name="use_gui" value="$(arg gui)" />

  <!-- Load controller configuration to parameter server -->
  <rosparam file="$(find akrobat)/config/akrobat_controllers.yaml" command="load" />
  <rosparam param="dynamixels">[m11, m12, m13, m21, m22, m23, m31, m32, m33, m41, m42, m43, m51, m52, m53, m61, m62, m63]</rosparam>

  <node respawn="true" pkg="joy" type="joy_node" name="akrobatJoy">
    <param name="dev" type="string" value="/dev/input/js0" />
  </node>

  <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher">
    <param name="publish_frequency" value="20.0" />
  </node>

  <node ns="$(arg dynamixel_namespace)" name="dynamixel_manager" pkg="dynamixel_controllers" type="controller_manager.py" required="true" output="screen">
    <rosparam>
      namespace: dxl_manager
      serial_ports:
        pan_tilt_port:
          port_name: "/dev/ttyUSB0"
          baud_rate: 200000.0
          min_motor_id: 1
          max_motor_id: 70
          update_rate: 20
    </rosparam>
  </node>

  <!-- Load joint controller configuration from YAML file to parameter server -->
  <rosparam ns="$(arg dynamixel_namespace)" file="$(find akrobat)/config/dynamixel_params.yaml" command="load"/>

  <!-- start specified joint controllers -->
  <node ns="$(arg dynamixel_namespace)" name="dynamixel_controller_spawner" pkg="dynamixel_controllers" type="controller_spawner.py"
  args="--manager=dxl_manager
        --port=pan_tilt_port
        --type=simple
        m11
        m12
        m13
        m21
        m22
        m23
        m31
        m32
        m33
        m41
        m42
        m43
        m51
        m52
        m53
        m61
        m62
        m63"
  output="screen">
  </node>

  <node ns="$(arg dynamixel_namespace)" name="dynamixel_joint_states_publisher" pkg="akrobat" type="$(find akrobat)src/akrobat/dynamixel_joint_state_publisher.py" output="screen">
    <param name="use_gui" value="TRUE" />
  </node>

  <node name="akrobat" pkg="akrobat" type="akrobat" output="screen" required="true">
  </node>
</launch>